models/experimental/functional_yolov4/reference/downsample1.py

```diff
@@ -0,0 +1,73 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+
+import torch
+import torch.nn as nn
+
+
+class DownSample1(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.c1 = nn.Conv2d(3, 32, 3, 1, 1, bias=False)
+        self.b1 = nn.BatchNorm2d(32)
+        self.relu = nn.ReLU(inplace=True)
+
+        self.c2 = nn.Conv2d(32, 64, 3, 2, 1, bias=False)
+        self.b2 = nn.BatchNorm2d(64)
+
+        self.c3 = nn.Conv2d(64, 64, 1, 1, 0, bias=False)
+        self.b3 = nn.BatchNorm2d(64)
+
+        self.c4 = nn.Conv2d(64, 64, 1, 1, 0, bias=False)
+        self.b4 = nn.BatchNorm2d(64)
+
+        self.c5 = nn.Conv2d(64, 32, 1, 1, 0, bias=False)
+        self.b5 = nn.BatchNorm2d(32)
+
+        self.c6 = nn.Conv2d(32, 64, 3, 1, 1, bias=False)
+        self.b6 = nn.BatchNorm2d(64)
+
+        self.c7 = nn.Conv2d(64, 64, 1, 1, 0, bias=False)
+        self.b7 = nn.BatchNorm2d(64)
+
+        self.c8 = nn.Conv2d(128, 64, 1, 1, 0, bias=False)
+        self.b8 = nn.BatchNorm2d(64)
+
+    def forward(self, input: torch.Tensor):
+        x1 = self.c1(input)
+        x1_b = self.b1(x1)
+        x1_m = self.relu(x1_b)
+
+        x2 = self.c2(x1_m)
+        x2_b = self.b2(x2)
+        x2_m = self.relu(x2_b)
+
+        x3 = self.c3(x2_m)
+        x3_b = self.b3(x3)
+        x3_m = self.relu(x3_b)
+
+        x4 = self.c4(x3_m)
+        x4_b = self.b4(x4)
+        x4_m = self.relu(x4_b)
+
+        x5 = self.c5(x4_m)
+        x5_b = self.b5(x5)
+        x5_m = self.relu(x5_b)
+
+        x6 = self.c6(x5_m)
+        x6_b = self.b6(x6)
+        x6_m = self.relu(x6_b)
+        x6_m = x6_m + x4_m
```

Comment:
@saichanda should we add x5_m and x3_m instead? Could you please take the model graph diagram and label it with the corresponding input and outputs, for instance the input to the very first conv labeled with "input" -> from the code. the output of it is x1 and so on. I have attached the graph for downsample1 here. we should do this for all the modules to ensure correct implementation. 

<img width="260" alt="Screenshot 2024-04-09 at 6 28 11 PM" src="https://github.com/tenstorrent-metal/tt-metal/assets/132954887/8fb2c6e1-065f-4883-a9a8-e0955cb8b1a7">

Reply:
![image](https://github.com/tenstorrent-metal/tt-metal/assets/164333733/9171518d-5f84-4ddd-a05c-67605fd7ad55)

Reply:
![downsample1_ref](https://github.com/tenstorrent-metal/tt-metal/assets/164333733/8871f7ce-b86c-4b15-93c8-9a026b61faeb)
![downsample1_ttnn](https://github.com/tenstorrent-metal/tt-metal/assets/164333733/2089a93d-56ca-4c5c-a766-a90c60ded0a3)